File: src/LCR193.er-cha-sou-suo-shu-de-zui-jin-gong-gong-zu-xian-lcof.100346/solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public TreeNode lowestCommonAncestor(TreeNode root, TreeNode p, TreeNode q) {
        if(p.val>q.val){
            TreeNode t = p;
            p = q;
            q = t;
        }
        return search(root,p,q);
    }
    
    private TreeNode search(TreeNode root, TreeNode p, TreeNode q){
        if(root == p || root == q){
            return root;
        }
        if(root.val<p.val){
            return search(root.right,p,q);
        }else if(root.val>q.val){
            return search(root.left,p,q);
        }
        return root;
    }
}

File: src/LCR193.er-cha-sou-suo-shu-de-zui-jin-gong-gong-zu-xian-lcof.100346/solution.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @param {TreeNode} p
 * @param {TreeNode} q
 * @return {TreeNode}
 */
var lowestCommonAncestor = function (root, p, q) {
    p = p.val;
    q = q.val;
    if (p > q) {
        const tmp = p;
        p = q;
        q = tmp;
    }

    return helper(root, p, q);
};

function helper (root, p, q) {
    if (!root || root.val === p || root.val === q) {
        return root;
    }
    if (root.val > p && root.val < q) {
        return root;
    } else if (root.val < p) {
        return helper(root.right, p, q);
    } else {
        return helper(root.left, p, q);
    }
}
